cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -0,0 +1,165 @@
+}
+
+mgp::Node GetStandinOrCopy(mgp::List &standin_nodes, mgp::Node node,
+                           std::map<mgp::Node, mgp::Node> &old_new_node_mirror) {
+  for (auto pair : standin_nodes) {
+    if (!pair.IsList() || !pair.ValueList()[0].IsNode() || !pair.ValueList()[1].IsNode()) {
+      throw mgp::ValueException(
+          "Configuration map must consist of specific keys and values described in documentation.");
+    }
+    if (node == pair.ValueList()[0].ValueNode()) {
+      return pair.ValueList()[1].ValueNode();
+    }
+  }
+  try {
+    return old_new_node_mirror.at(node);  // what if they send me wrong path
+  } catch (const std::out_of_range &e) {
+    throw mgp::ValueException("Can't clone relationship without cloning relationship's source and/or target nodes.");
+  }
+}
+
+bool CheckIfStandin(mgp::Node node, mgp::List standin_nodes) {
+  for (auto pair : standin_nodes) {
+    if (!pair.IsList() || !pair.ValueList()[0].IsNode()) {
+      throw mgp::ValueException(
+          "Configuration map must consist of specific keys and values described in documentation.");
+    }
+    if (node == pair.ValueList()[0].ValueNode()) {
+      return true;
+    }
+  }
+  return false;
+}
+
+void Refactor::CloneNodesAndRels(mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory,
+                                 const std::vector<mgp::Node> &nodes, const std::vector<mgp::Relationship> &rels,
+                                 const mgp::Map &config_map) {
+  mgp::List standin_nodes;
+  mgp::List skip_props;
+  if ((!config_map.At("standinNodes").IsList() && !config_map.At("standinNodes").IsNull()) ||
+      (!config_map.At("skipProperties").IsList() && !config_map.At("skipProperties").IsNull())) {
+    throw mgp::ValueException("Configuration map must consist of specific keys and values described in documentation.");
+  }
+  if (!config_map.At("standinNodes").IsNull()) {
+    standin_nodes = config_map.At("standinNodes").ValueList();
+  }
+  if (!config_map.At("skipProperties").IsNull()) {
+    skip_props = config_map.At("skipProperties").ValueList();
+  }
+  std::unordered_set<mgp::Value> skip_props_searchable{skip_props.begin(), skip_props.end()};
+
+  auto graph = mgp::Graph(memgraph_graph);
+
+  std::map<mgp::Node, mgp::Node> old_new_node_mirror;
+  for (auto node : nodes) {
+    if (CheckIfStandin(node, standin_nodes)) {
+      continue;
+    }
+    mgp::Node new_node = graph.CreateNode();
+
+    for (auto label : node.Labels()) {
+      new_node.AddLabel(label);
+    }
+
+    for (auto prop : node.Properties()) {
+      if (skip_props.Empty() || !skip_props_searchable.contains(mgp::Value(prop.first))) {
+        new_node.SetProperty(prop.first, prop.second);
+      }
+    }
+    old_new_node_mirror.insert({node, new_node});
+    InsertCloneNodesRecord(memgraph_graph, result, memory, node.Id().AsInt(), new_node.Id().AsInt());
+  }
```

Comment:
I think you can also put this in different function, which is just dealing with cloning nodes.

Reply:
I did it but I usually don't like doing that because I am just passing 1000 arguments around and it is rarely reusable since all the arguments are specific to that one procedure.. Is there maybe some other way to make it more readable (I'm guessing that is why you ask to do it), like surrounding it with a comment block or?